public_html/core/lib/config.php

```diff
@@ -184,37 +184,98 @@ private function _load_settings()
-                                           LIMIT 0,1)
+                      WHERE se.store_id = ".(int)$foundStoreId."
                             AND st.status = 1
                             AND TRIM(se.`group`) NOT IN
                                                     (SELECT TRIM(`key`) as `key`
                                                     FROM ".$db->table("extensions").")";
                 $query = $db->query($sql);
-                $store_settings = $query->rows;
+                $store_settings = [];
+                if($query->num_rows) {
+                    $store_settings = array_column($query->rows, 'value', 'key');
+                    $store_settings['store_id'] = $foundStoreId;
+                }
+
+                if($store_settings){
+                    //replace base url with url without fake(store-alias-path) subfolder
+                    $autoUri = rtrim($domain.dirname($_SERVER['SCRIPT_NAME']),'/');
+                    $autoUri .= $autoUri ? '/' : '';
+                    foreach(['config_url', 'config_ssl_url'] as $confUrl){
+                        $get =& $this->registry->get('request')->get;
+                        if($store_settings[$confUrl]) {
+                            $protocol = parse_url($store_settings[$confUrl], PHP_URL_SCHEME);
+                            if ($store_settings[$confUrl] != $protocol.'://'.$autoUri) {
+                                //remove fake(store-alias-path) prefix from seo-key which we got from .htaccess (_route_)
+                                $diff = substr( $store_settings[$confUrl], strlen($protocol.'://'.$autoUri));
+                                $store_settings['seo_prefix'] = $diff;
+                                    if(isset($get['_route_'])) {
+                                    $get['_route_'] = str_replace($diff, '', $get['_route_']);
+                                    if(!$get['_route_']){
+                                        unset($get['_route_']);
+                                    }
+                                }
+                                $store_settings[$confUrl] = $protocol.'://'.$autoUri;
+                            }
+                        }
+                    }
+                }
             }
 
             if ($store_settings) {
                 //store found by URL, load settings
-                foreach ($store_settings as $row) {
-                    $value = $row['value'];
-                    $this->cnfg[$row['key']] = $value;
-                }
+                $this->cnfg = array_merge($this->cnfg,$store_settings);
 
                 //fix for rare issue on a database and creation of empty cache
                 if ($this->cnfg['config_cache_enable']) {
                     $cache->push($cache_key, $store_settings);
                 }
-
-                $this->cnfg['config_store_id'] = $store_settings[0]['store_id'];
+                $this->cnfg['config_store_id'] = $store_settings['store_id'];
```

Comment:
Did we change how store setting are loaded? How do you access other stores in this array?

Reply:
Please ignore. I see it is local array